nucypher/policy/models.py

```diff
@@ -468,7 +467,55 @@ def __len__(self):
         return len(self.destinations)
 
 
-class WorkOrder(object):
+class RevocationNotice:
+    splitter = BytestringSplitter((bytes, 7),
+                                  (bytes, constants.NOTICE_LENGTH(32)),
+                                  Signature)
+    def __init__(self, node_id, arrangement_id, signature=None):
+        self.node_id = node_id
+        self.arrangement_id = arrangement_id
+        self.notice = b"REVOKE-" + arrangement_id
+        self.signature = signature or constants.NOT_SIGNED
+
+    def __bytes__(self):
+        if not self.signature is constants.NOT_SIGNED:
+            return self.notice + bytes(self.signature)
+        return constants.NOT_SIGNED
+
+    def __len__(self):
+        return len(b'REVOKE-') + len(self.arrangement_id)
+
+    @classmethod
+    def from_bytes(cls, some_bytes):
+        # TODO: Implement BytestringSplitter
+        _, arrangement_id, signature = cls.splitter(some_bytes)
+        return cls(None, arrangement_id, signature=signature)
```

Comment:
`None`?

Reply:
Yeah, it's never a good idea to do this. If this is really desired, I'd do `cls(node_id=None, ...)`... But should the `node_id` be there at all then?

Reply:
https://getyarn.io/yarn-clip/ba600782-d4a9-46be-8432-78f34cb9fc0c#SkHjKzkq3Q.copy